contracts/mocks/ComputedAddressRemainderMock.sol

```diff
@@ -0,0 +1,36 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+
+import {Clones} from "../proxy/Clones.sol";
+import {Create2} from "../utils/Create2.sol";
+
+contract ComputedAddressRemainderMock {
+    uint256 private constant _REMAINDER_MASK = ~(uint256(type(uint160).max));
+
+    function getCreate2ComputedRemainder(
+        bytes32 salt,
+        bytes32 bytecodeHash,
+        address deployer
+    ) external pure returns (uint256 remainder) {
+        address predicted = Create2.computeAddress(salt, bytecodeHash, deployer);
+        remainder = _getRemainder(predicted);
+    }
+
+    function getClonesPredictedRemainder(
+        address implementation,
+        bytes32 salt,
+        address deployer
+    ) external pure returns (uint256 remainder) {
+        address predicted = Clones.predictDeterministicAddress(implementation, salt, deployer);
+        remainder = _getRemainder(predicted);
+    }
+
+    function _getRemainder(address addr) internal pure returns (uint256 remainder) {
+        uint256 remainderMask = _REMAINDER_MASK;
+
+        assembly ("memory-safe") {
```

Comment:
We generally use the annotation syntax. No particular reason other than the Solidity docs stating that:

> In case you are developing a library that is meant to be compatible across multiple versions of Solidity, you can use a special comment to annotate an assembly block as memory-safe

```suggestion
        /// @solidity memory-safe-assembly
        assembly ("memory-safe") {
```

No big issue since it's a mock.